eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -13,3 +13,6 @@ steps:
         ScriptPath: eng/scripts/Set-VcpkgWriteModeCache.ps1
         azurePowerShellVersion: LatestVersion
         pwsh: true
+      # This step is idempotent and can be run multiple times in cases of
+      # failure and partial execution.
+      retryCountOnTaskFailure: 5
```

Comment:
Is this better than the inner retry via the ENV variable you were talking about?

Reply:
I would think 5 retries is a bit too high for this case. I imagine 3 is good enough, should we start there and see if we ever hit the case to ramp this up?

Otherwise, do we risk hitting 60 minute pipeline timeouts/other issues if we try to run this 5 times, with 4+ attempts failing?

Reply:
@weshaggard -- I don't think that we'll be able to retry on HTTP status codes because the error seen is not related to HTTP status responses. Unrouteable problems occur before HTTP. 

> For general HTTP response (except response status code is 429), Azure PowerShell uses the value defined in the AZURE_PS_HTTP_MAX_RETRIES environment variable. Its minimum value is 0. If not specified, Azure PowerShell uses the SDK default value.

Docs don't obviously show how to retry `Connect-AzAccount` in these routing failure scenarios. To fix it's easier to do retry on the task. We also don't get to control the parts of the `AzurePowerShell` task wrapping our own logic (the wrapping code is where the failure happened).

Reply:
@ahsonkhan -- I can't find docs describing how the retry delay is calculated but the delay does increase. 

> The failing task retries in seconds. The wait time between each retry increases after each failed attempt.

I'm fine with keeping it at 3.